pkg/providers/tinkerbell/template.go

```diff
@@ -467,6 +472,12 @@ func buildTemplateMapMD(clusterSpec *cluster.Spec, workerNodeGroupMachineSpec v1
 		"workerNodeGroupTaints":  workerNodeGroupConfiguration.Taints,
 	}
 
+	if clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].UpgradeRolloutStrategy != nil {
+		values["upgradeRolloutStrategy"] = true
+		values["maxSurge"] = clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].UpgradeRolloutStrategy.RollingUpdate.MaxSurge
+		values["maxUnavailable"] = clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].UpgradeRolloutStrategy.RollingUpdate.MaxUnavailable
+	}
```

Comment:
each worker node group can have different `UpgradeRolloutStrategy`
Let's move this to here https://github.com/aws/eks-anywhere/blob/3fc030e5cf750f5eb7a9f3b4ca647eb4313cfabf/pkg/providers/tinkerbell/template.go#L147-L150
so we can pass `UpgradeRolloutStrategy`  by worker node group